### PR TITLE
testing/traefik: upgrade to 2.0.4

### PR DESCRIPTION
**Pull request by andypost**
Ref https://blog.containo.us/traefik-2-0-6531ec5196c2

- https://docs.traefik.io/migration/v1-to-v2/
- https://docs.traefik.io/contributing/building-testing/

https://github.com/alpinelinux/aports/pull/11752

### CODE REVIEW

**Comment by andypost** [comment]
@joewholden is it time to upgrade?

File: testing/traefik/traefik.toml [hunk 1]
```diff
@@ -1,38 +1,27 @@
...
 
-defaultEntryPoints = ["http", "https"]
+[log]
+  level = "INFO"
 
 [entryPoints]
   [entryPoints.http]
   address = ":80"
-    [entryPoints.http.redirect]
-    entryPoint = "https"
```
Comment (andypost) [inline testing/traefik/traefik.toml]: There's no direct replacement but https://github.com/containous/traefik/issues/4863#issuecomment-534065126 works for me

File: testing/traefik/traefik.toml [hunk 2]
```diff
@@ -1,38 +1,27 @@
...
   [entryPoints.https]
   address = ":443"
-    [entryPoints.https.tls]
+  [entryPoints.traefik]
+  address = "localhost:8080"
```
Comment (andypost) [inline testing/traefik/traefik.toml]: it makes sense to expose api and other internal resources locally by default

File: testing/traefik/traefik.toml [hunk 3]
```diff
@@ -1,38 +1,27 @@
...
-#  onDemand = true
-#  onHostRule = true
-#  acmeLogging = true
-#  [acme.dnsChallenge]
+#  [certificatesResolvers.sample.acme.dnsChallenge]
 #    provider = "cloudflare"
 #    delayBeforeCheck = 10
-
-[rest]
-  entryPoint = "traefik"
```
Comment (andypost) [inline testing/traefik/traefik.toml]: I found no "rest" in v2

File: testing/traefik/traefik.toml [hunk 4]
```diff
@@ -1,38 +1,27 @@
...
+  [entryPoints.traefik]
+  address = "localhost:8080"
 
 [api]
+  insecure = true
+  dashboard = true
 
 [ping]
 
-#[consul]
```
Comment (andypost) [inline testing/traefik/traefik.toml]: there's support yet

File: testing/traefik/APKBUILD [hunk 5]
```diff
@@ -24,24 +25,27 @@ builddir="$srcdir/src/github.com/containous/$pkgname"
 #   - CVE-2019-9512
 #   - CVE-2019-9514
 
+export GOPATH="$srcdir"
+
 prepare() {
 	default_prepare
 	mkdir -p "$srcdir/src/github.com/containous"
 	mv "$srcdir"/$pkgname-$pkgver "$builddir"/
+#	CGO_ENABLED=0 GO111MODULE=off go get github.com/containous/go-bindata/...
```
Comment (andypost) [inline testing/traefik/APKBUILD]: It should be there according to https://docs.traefik.io/contributing/building-testing/#build-traefik

But it builds and works fine without this bindata

**Comment by andypost** [comment]
@joewholden I think better to close https://github.com/alpinelinux/aports/pull/11752 in favour of this one

**Comment by andypost** [comment]
https://github.com/containous/traefik/releases/tag/v2.0.4

File: testing/traefik/traefik.toml (hunk 4 above)
Comment (andypost) [reply testing/traefik/traefik.toml]: it was added in 2.0.4

**Comment by maxice8** [comment]
:shipit:

**Comment by andypost** [comment]
Next upgrade https://github.com/alpinelinux/aports/pull/12149

**Pull request closed without merging**